### DISCUSSION TITLE
Possible to have live preview of material included with the include directive?

### DISCUSSION THREAD

**Discussion by matthewfeickert** [jupyter-book/mystmd#1263] [Q&A]
[`myst start`](https://mystmd.org/guide/quickstart-myst-documents#start-myst) allows for live preview of the build of your document on save. This works when the changes are made to the main document (e.g. `main.md`) but if source files are included using the [include directive](https://mystmd.org/guide/directives#directive-include) (c.f. https://github.com/executablebooks/mystmd/discussions/1237) changes to them do not cause for a rebuild of the document. Instead, one will need to cause additional changes to the main document to then have the entire document rerendered.

When using `myst start`, is there a way to have source documents that are included through an include directive be watched for live rebuild? If so, how do you indicate this?

**Comment by rowanc1** [comment]
We do keep track of the dependencies, so this should be working as you describe. In my test just now, this line is shown on saving the included file `snippet.md`:

```text
🔄 Updating dependent pages for snippet.md [article.md]
📖 Built article.md in 90 ms.
```

However, the build process is only triggered if the included documents are in a folder that is being watched by myst, this generally means they are in/underneath the folder containing the `myst.yml`. What is the folder layout of your included file and myst project?

**Reply by matthewfeickert** [reply]
> What is the folder layout of your included file and myst project?

```console
$ tree paper/
paper/
├── acknowledgements.md
├── banner.png
├── conclusions.md
├── figures
│   ├── figure1.png
│   └── figure2.png
├── introduction.md
├── main.md
├── mybib.bib
├── myst.yml
├── scientific-python-ecosystem.md
└── uncovering-higgs.md
```

where 

```console
$ tail -n 14 paper/main.md 
:::{include} introduction.md
:::

:::{include} scientific-python-ecosystem.md
:::

:::{include} uncovering-higgs.md
:::

:::{include} conclusions.md
:::

:::{include} acknowledgements.md
:::
```

For `myst` `v1.2.5`, if I run `myst start` in the `paper` directory

```
🪄  Linked 1 DOI in 3.1 ms for main.md
📖 Built main.md in 253 ms.
📚 Built 1 page for project in 335 ms.


	✨✨✨  Starting Article Theme  ✨✨✨



🔌 Server started on port 3000!  🥳 🎉


	👉  http://localhost:3000  👈


💌 GET  / 200 - - 103.284 ms
💌 GET  /myst-theme.css 200 - - 5.407 ms
```

and then make a change to `introduction.md`

```
$ git diff paper/introduction.md
diff --git a/paper/introduction.md b/paper/introduction.md
index a764d85..9f5108c 100644
--- a/paper/introduction.md
+++ b/paper/introduction.md
...
+
+A change was made.
```

there is no build log update from `myst start`.

Repository here: https://github.com/matthewfeickert/draft-scipy-2024-proceedings-submission

**Reply by rowanc1** [reply]
This all very much looks like it should work, we will investigate in the next few days (post SciPy deadline though!). cc @fwkoch. Hope the writing is going smoothly otherwise. :)

**Reply by matthewfeickert** [reply]
Thanks @rowanc1 and SGTM. :+1: To be clear, live rebuild on changes to `paper/main.md` works as expected. It is only changes to included files where the live rebuild isn't working.
